### PR TITLE
Unit tests PPA (EXPOSUREAPP-6723)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Just tests and some bug fixes ( clear/reset was at the wrong spot, test result collector logic)

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt [hunk 1]
```diff
@@ -99,24 +99,31 @@ class AnalyticsTestResultCollector @Inject constructor(
...
     }
 
-    fun updatePendingTestResultReceivedTime(testResult: CoronaTestResult, type: CoronaTest.Type) {
+    fun reportTestResultReceived(testResult: CoronaTestResult, type: CoronaTest.Type) {
         if (analyticsDisabled) return
-        val shouldUpdate = type.settings.testScannedAfterConsent.value &&
-            type.settings.testResultAtRegistration.value.isPending &&
-            testResult.isFinal
-        if (shouldUpdate) {
+        if (testResult.isFinal) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt:123]: Same as above,  `isFinal`  or `!isPending`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultDonor.kt [hunk 2]
```diff
@@ -30,12 +30,6 @@ abstract class AnalyticsTestResultDonor(
 ) : DonorModule {
 
     override suspend fun beginDonation(request: DonorModule.Request): DonorModule.Contribution {
-        val scannedAfterConsent = testResultSettings.testScannedAfterConsent.value
-        if (!scannedAfterConsent) {
-            Timber.d("Skipping TestResultMetadata donation (scannedAfterConsent=%s)", scannedAfterConsent)
-            return TestResultMetadataNoContribution
-        }
-
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultDonor.kt:38]: Why is the check not necessary anymore?

**Review by d4rken — COMMENTED** [review]
Looks good, few comments.
Nice unit tests :+1:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt [hunk 3]
```diff
@@ -99,24 +99,31 @@ class AnalyticsTestResultCollector @Inject constructor(
 
         if (testResult !in validTestResults) return // Not interested in other values
 
+        type.settings.testResultAtRegistration.update { testResult }
+
+        if (testResult.isFinal) {
+            type.settings.finalTestResultReceivedAt.update { timeStamper.nowUTC }
+        }
```
[the remark this answers is not in the document]
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt:106]: That's a different implementation you have in mind. Here isFinal is true when test is positive or negative

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt:123]: same here

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultDonor.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultDonor.kt:38]: I changed the logic to be more like the submission key implementation. The testRegisteredAt is only set when there is consent at registration. That field is checked here anyway, so scannedAfterConsent is not necessary any more.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/testresult/AnalyticsTestResultCollector.kt:106]: Ah, yeah I confused the two. Maybe we should rename one of them?

**Review by d4rken — DISMISSED** [review]

**Review by mtwalli — APPROVED** [review]

**Review by d4rken — APPROVED** [review]

**Pull request merged**